### ISSUE TITLE
Clustermap with square=True disorganise the dendrogram and the heatmap

### ISSUE THREAD

**Issue by EmanuelGoncalves** [mwaskom/seaborn#483]

**Comment by mwaskom** [comment]
Can you show an example so I can understand what you mean by "disorganise"?

**Comment by EmanuelGoncalves** [comment]
![clustermap](https://cloud.githubusercontent.com/assets/3500746/6757201/2a096184-cf27-11e4-8ee5-6dc49b7e7ec3.png)

**Comment by mwaskom** [comment]
Thanks.

I cannot reproduce this bug locally, but I also don't know exactly what you are trying to do. It would be most helpful to include:

- The code you used to get here. If you can reproduce the issue using randomly generated data, so that I can copy and paste and try to reproduce on my own system, that would be best. But even just seeing what options you're using the produce this result would help.
- The version of seaborn that you're using (`seaborn.__version__`), and ideally the matplotlib version too.

Also is this a saved figure or a screenshot from an IPython notebook output?

**Comment by EmanuelGoncalves** [comment]
Thanks you for your support.

I've to use TkAgg backend as other backends crash on my laptop. The image I've showed you was saved as png.

Details: Mac OS 10.10.2, Python 2.7.6, Seaborn 0.5.1, Matplotlib 1.4.2.

```python
import matplotlib
matplotlib.use('TkAgg')

import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

data = np.random.rand(10, 5)
sns.clustermap(data, square=True)
plt.savefig('example.pdf')
```
![example](https://cloud.githubusercontent.com/assets/3500746/6764624/a1bc6048-cfb1-11e4-80ac-355fb3085515.png)

**Comment by JWarmenhoven** [comment]
It appears that the plotting of the dendrograms does not take into account the keyword 'square'. I think this keyword is just passed on to the heatmap function after the dendrograms have been defined. Plotting of col_colors and row_colors in combination with 'square=True' therefore also does not produce the expected result.

````python
data = np.random.rand(5, 3)

colors = sns.color_palette('Blues', n_colors=3)
col_colors = pd.Series([colors[0], colors[1], colors[2]])

sns.clustermap(data, square=True, col_colors=col_colors)

````
![index](https://cloud.githubusercontent.com/assets/7450204/6994992/2b8e96e8-db2d-11e4-8d8b-140dfd1502b5.png)


````python
sns.clustermap(data, square=False, col_colors=col_colors)
````
![index](https://cloud.githubusercontent.com/assets/7450204/6994991/169ba9a6-db2d-11e4-9e2d-2585f442a0dd.png)


Seaborn 0.6dev

**Comment by mwaskom** [comment]
This behavior makes sense, actually. The `square=` keyword argument is handled within `heatmap` and uses the matplotlib Axes `aspect` property, which saves a lot of tricky math on the seaborn end. So this isn't so much a "simple bug" (though I can see why you'd want it to work properly) but rather a behavior that isn't fully supported and would take some effort to get working in clustermap.

**Comment by mwaskom** [comment]
Closing as I think this kind of thing will be addressed through #441
